tests/test_process_arbiter.py

```diff
@@ -0,0 +1,165 @@
+        package_name="dummy",
+        version="0.0.0.dev",
+        notify_failure=True,
+        notify_retry=True,
+        env={"KIRBY_TEST_MARKER": "hello, world!"},
+    )
+    assert arbiter.venv_name == "kirby-dummy-0.0.0.dev"
+
+
+@pytest.mark.skipif(
+    not os.getenv("DUMMY_PACKAGE_INSTALL"),
+    reason=(
+        "You must pass the name of a PyPi package "
+        "to install with DUMMY_PACKAGE_INSTALL"
+    ),
+)
+@pytest.mark.skipif(
+    not os.getenv("DUMMY_PACKAGE_VERSION"),
+    reason=(
+        "You must pass the *exact* version of a PyPi package "
+        "to install with DUMMY_PACKAGE_VERSION"
+    ),
+)
+def test_arbiter_can_ensure_virtualenv_creation(venv_directory):
+    arbiter = Arbiter(
+        script_type=JobType.SCHEDULED,
+        package_name=os.getenv("DUMMY_PACKAGE_INSTALL"),
+        version=os.getenv("DUMMY_PACKAGE_VERSION"),
+        notify_failure=True,
+        notify_retry=True,
+        env={"KIRBY_TEST_MARKER": "hello, world!"},
+    )
+
+    context = arbiter.ensure_environment(venv_directory)
+
+    assert (
+        os.getenv("DUMMY_PACKAGE_INSTALL"),
+        os.getenv("DUMMY_PACKAGE_VERSION"),
+    ) in context.installed_packages
+
+
+@pytest.mark.skipif(
+    not os.getenv("DUMMY_PACKAGE_INSTALL"),
+    reason=(
+        "You must pass the name of a PyPi package "
+        "to install with DUMMY_PACKAGE_INSTALL"
+    ),
+)
+@pytest.mark.skipif(
+    not os.getenv("DUMMY_PACKAGE_VERSION"),
+    reason=(
+        "You must pass the *exact* version of a PyPi package "
+        "to install with DUMMY_PACKAGE_VERSION"
+    ),
+)
+def test_arbiter_can_start_scheduled_process(venv_directory):
+    arbiter = Arbiter(
+        script_type=JobType.SCHEDULED,
+        package_name=os.getenv("DUMMY_PACKAGE_INSTALL"),
+        version=os.getenv("DUMMY_PACKAGE_VERSION"),
+        notify_failure=True,
+        notify_retry=True,
+        env={"KIRBY_TEST_MARKER": "hello, world!"},
+    )
+
+    arbiter.run()
+    assert arbiter.status == ProcessState.RUNNING
+
+
+@pytest.mark.skipif(
+    not os.getenv("DUMMY_PACKAGE_INSTALL"),
+    reason=(
+        "You must pass the name of a PyPi package "
+        "to install with DUMMY_PACKAGE_INSTALL"
+    ),
+)
+@pytest.mark.skipif(
+    not os.getenv("DUMMY_PACKAGE_VERSION"),
+    reason=(
+        "You must pass the *exact* version of a PyPi package "
+        "to install with DUMMY_PACKAGE_VERSION"
+    ),
+)
+def test_arbiter_can_start_daemon_process(venv_directory):
+    arbiter = Arbiter(
+        script_type=JobType.DAEMON,
+        package_name=os.getenv("DUMMY_PACKAGE_INSTALL"),
+        version=os.getenv("DUMMY_PACKAGE_VERSION"),
+        notify_failure=True,
+        notify_retry=True,
+        env={"KIRBY_TEST_MARKER": "hello, world!"},
+    )
+    arbiter.run()
+    assert arbiter.status == ProcessState.RUNNING
+
+    arbiter.join()
+    assert arbiter.status == ProcessState.STOPPED
+
+
+# def test_arbiter_can_restart_daemon_process(venv_directory):
```

Comment:
I'm not too fond of commented out code, either delete it or fix it 😄

Reply:
I just renamed the original file. I'll submit my work on that on the future commits.

 https://github.com/adimian/kirby/pull/39/commits/5a08163f367fdd50d9cdab7ef43fdafc7b2e52dd#diff-42e46e928ccad6d9687486311b369bcdR81